ai/mistralai/v0/tasks.go

```diff
@@ -0,0 +1,158 @@
+
+type ChatMessage struct {
+	Role    string              `json:"role"`
+	Content []MultiModalContent `json:"content"`
+}
+type URL struct {
+	URL string `json:"url"`
+}
+
+type MultiModalContent struct {
+	ImageURL URL    `json:"image-url"`
+	Text     string `json:"text"`
+	Type     string `json:"type"`
+}
+
+type textGenerationInput struct {
+	ChatHistory  []ChatMessage `json:"chat-history"`
+	MaxNewTokens int           `json:"max-new-tokens"`
+	ModelName    string        `json:"model-name"`
+	Prompt       string        `json:"prompt"`
+	PromptImages []string      `json:"prompt-images"`
+	Seed         int           `json:"seed"`
+	SystemMsg    string        `json:"system-message"`
+	Temperature  float64       `json:"temperature"`
+	TopK         int           `json:"top-k"`
+	TopP         float64       `json:"top-p"`
+	Safe         bool          `json:"safe"`
+}
+
+type chatUsage struct {
+	InputTokens  int `json:"input-tokens"`
+	OutputTokens int `json:"output-tokens"`
+}
+
+type textGenerationOutput struct {
+	Text  string    `json:"text"`
+	Usage chatUsage `json:"usage"`
+}
+
+type textEmbeddingInput struct {
+	Text          string `json:"text"`
+	ModelName     string `json:"model-name"`
+	EmbeddingType string `json:"embedding-type"`
+}
+
+type textEmbeddingUsage struct {
+	Tokens int `json:"tokens"`
+}
+
+type textEmbeddingOutput struct {
+	Embedding []float64          `json:"embedding"`
+	Usage     textEmbeddingUsage `json:"usage"`
+}
+
+func (e *execution) taskTextGeneration(in *structpb.Struct) (*structpb.Struct, error) {
+
+	inputStruct := textGenerationInput{}
+	err := base.ConvertFromStructpb(in, &inputStruct)
+	if err != nil {
+		return nil, fmt.Errorf("error generating input struct: %v", err)
+	}
+
+	messages := []mistralSDK.ChatMessage{}
+
+	if inputStruct.SystemMsg != "" {
+		messages = append(messages, mistralSDK.ChatMessage{
+			Role:    "system",
+			Content: inputStruct.SystemMsg,
+		})
+	}
+	for _, chatMessage := range inputStruct.ChatHistory {
+		messageContent := ""
+		for _, content := range chatMessage.Content {
+			if content.Type == "text" {
+				messageContent += content.Text
+			}
+		}
+		if messageContent == "" {
+			continue
+		}
+		messages = append(messages, mistralSDK.ChatMessage{
+			Role:    chatMessage.Role,
+			Content: messageContent,
+		})
+	}
+
+	promptMessage := mistralSDK.ChatMessage{
+		Role:    "user",
+		Content: inputStruct.Prompt,
+	}
+
+	messages = append(messages, promptMessage)
+
+	params := mistralSDK.ChatRequestParams{
+		Temperature: inputStruct.Temperature,
+		RandomSeed:  inputStruct.Seed,
+		MaxTokens:   inputStruct.MaxNewTokens,
+		TopP:        inputStruct.TopP,
+		SafePrompt:  inputStruct.Safe,
+	}
+
+	resp, err := e.client.sdkClient.Chat(
+		inputStruct.ModelName,
+		messages,
+		&params,
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("error calling Chat: %v", err)
+	}
+
+	outputStruct := textGenerationOutput{}
+
+	outputStruct.Text = resp.Choices[len(resp.Choices)-1].Message.Content
```

Comment:
Question
Does it mean Mistral responds several messages.
And, here, you decide to choose the latest one. right?
Is there any reference about this part I can check as well?
Thank you!
```
outputStruct.Text = resp.Choices[len(resp.Choices)-1].Message.Content
```

Reply:
I originally thought this has to do with multi-turn conversation. However, after some testing, the length of choices is always 1. The following script yield length=1.

```
model = "mistral-large-latest"

client = MistralClient(api_key=api_key)

messages = [
    ChatMessage(role="user", content="I am a chinese user, pleass answer me in Chinese."),
    ChatMessage(role="assistant", content="沒問題，我會用中文回答你。"),
    ChatMessage(role="user", content="What is the capital of France?"),
]

# No streaming
chat_response = client.chat(
    model=model,
    messages=messages,
    temperature=1.0
)

print(len(chat_response.choices))


print(chat_response.choices[0].message.content)
```

And the documentation doesn't specify the meaning of its property, I will change to always return choices[0]'s value.

Reply:
OK! Seems not to be a blocker. 
Please test more cases, such as different models, to make sure it is ok.

Memo
In Doc, it seems to be designed as multiple.
![image](https://github.com/user-attachments/assets/d24969d2-b75e-473f-90c8-9581241b953a)

However, in their sample code, they also collect the 0.
![image](https://github.com/user-attachments/assets/7edf4d00-8081-4d2f-acbb-b7ba910835e8)